addon/-private/functional/modifier.ts

```diff
@@ -89,26 +154,28 @@ export default function modifier(
   fn: (
     element: Element,
     positional: unknown[],
-    named: Record<string, undefined>
-  ) => void | Teardown
+    named: object
+  ) => void | Teardown,
+  options?: { eager: boolean }
```

Comment:
Can we default here?
```suggestion
  options?: { eager: boolean } = { eager: true }
```

This makes it a bit clearer what the current default is (to my eyes at least)

Reply:
Ah! Also lets us simplify the body a bit, too: it can just become a check on `options.eager` directly.